server/hwapi/external/c3/client.py

```diff
@@ -44,6 +47,77 @@ class C3Client:
 
     def __init__(self, db: Session):
         self.db = db
+        self.session = self._create_session_with_retries()
+
+    def _create_session_with_retries(self):
+        """Create a requests session with retry strategy"""
+        session = requests.Session()
+
+        retry_strategy = Retry(
+            total=5,  # Total number of retries
+            backoff_factor=2,  # Wait time between retries (2**retry_count seconds)
+            status_forcelist=[429, 500, 502, 503, 504],  # HTTP status codes to retry
+            allowed_methods=["HEAD", "GET", "OPTIONS"],  # Only retry safe methods
+            raise_on_status=False,  # Don't raise exception on retry-able status codes
+        )
+
+        # Mount adapter with retry strategy
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        session.mount("http://", adapter)
+        session.mount("https://", adapter)
+
+        return session
+
+    def _make_request_with_retries(
+        self, url: str, max_retries=5, base_delay=2, max_delay=60
+    ):
+        """Make HTTP request with custom retry logic for timeout errors"""
+        for attempt in range(max_retries):
+            try:
+                logger.debug(
+                    f"Attempting request to {url} (attempt {attempt + 1}/{max_retries})"
+                )
+                response = self.session.get(url, timeout=90)
+                response.raise_for_status()
+                return response
+
+            except (
+                requests.exceptions.ReadTimeout,
+                requests.exceptions.ConnectTimeout,
+                requests.exceptions.ConnectionError,
+            ) as e:
+
+                if attempt == max_retries - 1:  # Last attempt
+                    logger.error(f"Failed to fetch {url} after {max_retries} attempts")
+                    raise e
+
+                # Calculate delay with exponential backoff
+                delay = min(base_delay * (2**attempt), max_delay)
+                logger.warning(
+                    f"Request to {url} failed (attempt {attempt + 1}/{max_retries}): {str(e)}. "
+                    f"Retrying in {delay} seconds..."
+                )
+                time.sleep(delay)
+
+            except requests.exceptions.HTTPError as e:
+                # For HTTP errors, check if it's worth retrying
+                if e.response.status_code in [429, 500, 502, 503, 504]:
```

Comment:
I think this is a little more readable, let me know what you think:


```suggestion
                status_code = HTTPStatus(e.response.status_code)
                if status_code.is_server_error or status_code == HTTPStatus.TOO_MANY_REQUESTS:
```

The above needs the following import:

```python
from http import HTTPStatus
```